specs/docs/security/DataHubSecurity.md

```diff
@@ -0,0 +1,266 @@
+
+#### Associate Authorities
+ - canLoginToHubCentral
+
+#### Inherited Roles
+ - data-hub-common
+
+### hub-central-downloader
+
+Role to determine if a user can download the Hub Central project.
+
+#### Associate Authorities
+ - canDownloadProject
+
+#### Inherited Roles
+ - data-hub-common
+
+### hub-central-clear-data
+
+Role to determine if a user can clear data through Hub Central.
+
+#### Associate Authorities
+ - canClearData
+
+#### Inherited Roles
+ - data-hub-common-writer
+
+### hub-central-step-runner
+
+Can run steps through Hub Central
+
+#### Associate Authorities
+ - canReadStep
+ - canRunStep
+
+#### Inherited Roles
+ - hub-central-user
+ - data-hub-common-writer
+ - data-hub-flow-reader
+ - data-hub-load-data-reader
+ - data-hub-mapping-reader
+ - data-hub-match-merge-reader
+ - data-hub-step-definition-reader
+
+### hub-central-step-writer
+
+Can create and run steps though Hub Central
+
+#### Associate Authorities
+ - canWriteStep
+
+#### Inherited Roles
+ - hub-central-step-runner
+ - data-hub-flow-writer
+
+### hub-central-load-reader
+
+Can read load data artifacts through Hub Central
+
+#### Associate Authorities
+ - canReadLoadData
+
+#### Inherited Roles
+ - hub-central-user
+ - data-hub-load-data-reader
+
+### hub-central-load-writer
+
+Can read and write load data artifacts through Hub Central
+
+#### Associate Authorities
+ - canWriteLoadData
+
+#### Inherited Roles
+ - hub-central-load-reader
+ - data-hub-common-writer
+ - data-hub-load-data-reader
+
+### hub-central-mapping-reader
+
+Can read mapping artifacts through Hub Central
+
+#### Associate Authorities
+ - canReadMapping
+
+#### Inherited Roles
+ - hub-central-user
+ - data-hub-mapping-reader
+
+### hub-central-mapping-writer
+
+Can read and write mapping artifacts through Hub Central
+
+#### Associate Authorities
+ - canWriteMapping
+
+#### Inherited Roles
+ - hub-central-mapping-reader
+ - data-hub-common-writer
+ - data-hub-mapping-writer
+
+### hub-central-match-merge-reader
+
+Can read match and merge artifacts through Hub Central
+
+#### Associate Authorities
+ - canReadMatching
+ - canReadMerge
+
+#### Inherited Roles
+ - hub-central-user
+ - data-hub-match-merge-reader
+
+### hub-central-match-merge-writer
+
+Can read and write match and merge artifacts through Hub Central
+
+#### Associate Authorities
+ - canWriteMatching
+ - canWriteMerge
+
+#### Inherited Roles
+ - hub-central-match-merge-reader
+ - data-hub-common-writer
+ - data-hub-match-merge-writer
+
+### hub-central-custom-reader
+
+Can read custom artifacts through Hub Central
+
+#### Associate Authorities
+ - canReadCustom
+
+#### Inherited Roles
+ - hub-central-user
+ - data-hub-step-definition-reader
+
+### hub-central-custom-writer
+
+Can read and write custom artifacts through Hub Central
+
+#### Associate Authorities
+ - canWriteCustom
+
+#### Inherited Roles
+ - hub-central-mapping-reader
+ - data-hub-common-writer
+ - data-hub-step-definition-writer
+ - data-hub-modules-writer
```

Comment:
Should hub-central-custom-writer inherit hub-central-custom-reader, so we can do away with inheriting data-hub-step-definition-writer. 
Typo in modules-writer, should be data-hub-module-writer. Should data-hub-common-writer inherit data-hub-module-writer?